### ISSUE TITLE
FP in 'Key Vault Logging Disabled' query for Terraform

### ISSUE THREAD

**Issue by t3mi** [Checkmarx/kics#3440]
### Platform
Terraform

### Provider
Azure

### Description
Following code produces false positive result and maybe it's a duplicate for 'Vault Auditing Disabled' query.
```hcl
resource "azurerm_key_vault" "kv" {
  name                            = "veryuniquekv"
  location                        = azurerm_resource_group.rg.location
  resource_group_name             = azurerm_resource_group.rg.name
  tenant_id                       = data.azurerm_client_config.current.tenant_id
  sku_name                        = "standard"
  enabled_for_disk_encryption     = true
  enabled_for_deployment          = true
  enabled_for_template_deployment = true
  purge_protection_enabled        = false
}

resource "azurerm_monitor_diagnostic_setting" "kv" {
  name                       = "AuditLogs"
  target_resource_id         = azurerm_key_vault.kv.id
  log_analytics_workspace_id = azurerm_log_analytics_workspace.oms.id
  
  log {
    category = "AuditEvent"

    retention_policy {
      enabled = false
    }
  }

  metric {
    category = "AllMetrics"

    retention_policy {
      enabled = false
    }
  }
}
```
